datafusion/sql/src/expr/function.rs

```diff
@@ -84,73 +86,88 @@ impl<'a, S: ContextProvider> SqlToRel<'a, S> {
             } else {
                 WindowFrame::new(!order_by.is_empty())
             };
-            let fun = self.find_window_func(&name)?;
-            let expr = match fun {
-                WindowFunction::AggregateFunction(aggregate_fun) => {
-                    let (aggregate_fun, args) = self.aggregate_fn_to_expr(
-                        aggregate_fun,
-                        function.args,
-                        schema,
-                        planner_context,
-                    )?;
-
-                    Expr::WindowFunction(expr::WindowFunction::new(
-                        WindowFunction::AggregateFunction(aggregate_fun),
-                        args,
+            if let Ok(fun) = self.find_window_func(&name) {
```

Comment:
The diff display on GitHub is not clear, it's just a simple refactor
Before:
```rust
if let Some(WindowType::WindowSpec(window)) = function.over.take() {
		/* 1 */
		let fun = self.find_window_func(&name)?;
		/* 2 */
		return ...;
}

/* 3 */

// Return error
Err(...)
```
After:
```rust
if let Some(WindowType::WindowSpec(window)) = function.over.take() {
		/* 1 */
		if let Ok(fun) = self.find_window_func(&name) {
				/* 2 */
				return ...;
		}
} else {
		/* 3 */
}

// Return error
Err(...)
```

Reply:
The whitespace blind diff also shows this nicely: https://github.com/apache/arrow-datafusion/pull/6520/files?w=1